git.tex

```diff
@@ -1,11 +1,38 @@
-\subsection{Development Model}
+\subsection{Development Model}\label{sec:development}
 
-Lots of small repositories.
+To enable concurrent development across all six LSST sites, the LSST development team uses a decentralized development model based on the Git version control system\footnote{\url{https://git-scm.com/}}.
+[TODO: describe in terms of Agile values?]
 
-Feature branches associated with Jira tickets.
+\subsubsection{Version Control}\label{sec:git}\label{sec:subversion}
 
-Branch protection.
+[TODO: why did LSST switch from Subversion to Git?]
 
-Rebase before merging.
+\subsubsection{Code Organization}\label{sec:git_repositories}
+
+The data management system is divided into several hundred individual Git repositories, hosted on GitHub\footnote{\url{http://github.com/}}.
+Each repository corresponds to one code package, and
+all but a handful of repositories usually have at most one developer working on them at a time.
+Isolating code in small repositories minimizes the need for developers to frequently download code updates, something the Git framework requires whenever developers make concurrent changes to the \texttt{master} branch of the same repository.
+Representing each package by an independent repository also makes it easy to formalize and track inter-package dependencies using tools like EUPS (\autoref{sec:eups}).
+
+\subsubsection{The LSST Workflow}\label{sec:dev_workflow}
+
+LSST software development is based on the shared repository model:
+work is done on branches of a single repository rather than on user forks.
+Development work is associated with a Jira ticket (\autoref{sec:jira_ticket}), and
+keeping all work in a shared repository makes it easier to link code changes to their corresponding Jira tickets.
+[TODO: why is that important? I assume it has something to do with earned value tracking]
```

Comment:
I think this relates to us keeping branches around and not deleting them. With forks the branch never turns up in the parent repo. It's easier to audit the work later if the branch and the Jira ticket are related: you can work it out if the branch was deleted but it depends on you parsing the merge commits looking for the branch name. @ktlim has suggested that deleting merged branches is okay so long as we replace them with tags. Also, Jira can't see branches on forks but can see branches on the parent repo (it may be that we don't care about that but it is sometimes useful to be able to go to a branch directly from Jira even if no PR has been made yet).

Reply:
I guess I'm not sure exactly what "audit the work" means, since it's not something I see.

Reply:
I don't think anyone has ever asked to audit a story. It is the case that Jira won't see forks and we want Jira to see the branches.